ConfigSpace/hyperparameters.pyx

```diff
@@ -410,6 +522,52 @@ cdef class IntegerHyperparameter(NumericalHyperparameter):
 
     cpdef np.ndarray _transform_vector(self, np.ndarray vector):
         raise NotImplementedError()
+    
+    def pdf(self, vector: np.ndarray) -> np.ndarray:
+        """
+        Computes the probability density function of the parameter in 
+        the original parameter space (the one specified by the user).
+        For each parameter type, there is also a method _pdf which 
+        operates on the transformed (and possibly normalized) parameter
+        space. Only legal values return a positive probability density,
+        otherwise zero.
+        
+        Parameters
+        ----------
+        vector: np.ndarray
+            the (N, ) vector of imputs for which the probability density
+            function is to be computed.
+        """
+        if vector.ndim != 1:
+            raise ValueError("Method pdf expects a one-dimensional numpy array")
+        is_integer = (np.round(vector) == vector).astype(int)
+        vector = self._inverse_transform(vector)
+        return self._pdf(vector) * is_integer
```

Comment:
Yes. This is not paramount for any of the internals of PiBO to work, but it seemed like the correct thing to do. Do you want to change it?

(note that this would go away if we have a uniform pdf function. it can **not** be moved to _pdf.